[feat] Add user details edit function and profile image policies

IiamV opened the pull request:
## What?

Implemented the `edit_user_details` SQL function and created 4 new storage policies for the `profile-images` bucket.

* **Function: `edit_user_details`**

  * **Overview**: Updates user information in the `user_details` table to support and optimize the `updateUserProfile` Edge Function.
  * **Input**:

    ```sql
    p_user_id uuid,
    p_full_name text,
    p_birth_date date
    ```
  * **Output**: `Boolean`

* **4 New Policies**

  * **Overview**: Grants public access to view, upload, update, and delete profile images in the `profile-images` bucket.

## Why?

To let users update their profile information..

## How?

* Added a SQL RPC function `edit_user_details` for partial updates based on user input.
* Created 4 new RLS policies on the `profile-images` bucket to allow public operations: `select`, `insert`, `update`, and `delete`.

IiamV committed:
Add user details edit function and profile image policies

Create the edit_user_details function to update user information in the user_details table to assist the updateUserProfile edge function. Adds policies to allow anyone to view, upload, update, and delete profile images in the 'profile-images' bucket.

IiamV commented:
> @maik205 

These are the necessary database policies and function needed for the updateUserProfile edge function.

maik205 commented:
Can you make the RLS policies tighter? (ie. only allowing updates, deletes on user-owned storage objects)

IiamV committed:
Restrict profile image updates and deletes to owners

The policies now check that the user is authenticated and that the object name contains their user ID.

IiamV commented:
@maik205 
> Can you make the RLS policies tighter? (ie. only allowing updates, deletes on user-owned storage objects)

Does the latest commit look good?

maik205 commented on policy/user_update_profile.sql:
```diff
@@ -9,12 +8,20 @@ ON storage.objects
+USING (
+  bucket_id = 'profile-images' 
+  AND auth.uid() IS NOT NULL
+  AND name LIKE '%' || auth.uid()::text || '%'
```
you should check the ownership of the file with the owner field in DB.

maik205 commented on policy/user_update_profile.sql:
```diff
@@ -9,12 +8,20 @@ ON storage.objects
-USING (bucket_id = 'profile-images');
+USING (
+  bucket_id = 'profile-images' 
+  AND auth.uid() IS NOT NULL
```
Please apply check `authenticated` database role instead of manual checking to avoid jwt overhead for this function

IiamV replied:
> @maik205 

I've change it to
```plpgsql
TO authenticated
USING (
  bucket_id = 'profile-images' 
  AND owner = auth.uid()
);
```

maik205 replied:
Nice!

IiamV committed:
Update profile image policies to use owner field

Changed the update and delete policies for profile images to check if the authenticated user is the owner, instead of matching user ID in the object name.
